### PR TITLE
Add files and document the nix experiment

### PR DESCRIPTION
**Pull request by ahobson**
## Description

To allow more folks to join the nix experiment, add the config files and document how to start the experiment.

## Reviewer Notes

This shouldn't change anything for folks that don't have nix installed.

## Code Review Verification Steps

* [ ] If the change is risky, it has been tested in experimental before merging.
* [ ] Code follows the guidelines for [Logging](https://github.com/transcom/mymove/wiki/Backend-Programming-Guide#logging)
* [ ] The requirements listed in [Querying the Database Safely](https://github.com/transcom/mymove/wiki/Backend-Programming-Guide#querying-the-database-safely) have been satisfied.
* Any new migrations/schema changes:
  * [ ] Follow our guidelines for zero-downtime deploys (see [Zero-Downtime Deploys](https://github.com/transcom/mymove/wiki/migrate-the-database#zero-downtime-migrations))
  * [ ] Have been communicated to #g-database
  * [ ] Secure migrations have been tested following the instructions in our [docs](https://github.com/transcom/mymove/wiki/migrate-the-database#secure-migrations)
* [ ] There are no aXe warnings for UI.
* [ ] This works in [Supported Browsers and their phone views](https://github.com/transcom/mymove/tree/master/docs/adr/0016-Browser-Support.md) (Chrome, Firefox, IE, Edge).
* [ ] Tested in the Experimental environment (for changes to containers, app startup, or connection to data stores)
* [ ] User facing changes have been reviewed by design.
* [ ] Request review from a member of a different team.
* [ ] Have the Jira acceptance criteria been met for this change?

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -341,6 +345,25 @@ If you want to develop against AWS services you will need an AWS user account wi
...
+1. Ensure you have run `direnv allow` to set up the appropriate nix
+   environment variables.
+1. Make sure you have disabled any `nodeenv`, `asdf` or any other
+   version switchers for mymove.
+1. Run `./nix/update.sh`
+
+If the nix dependencies change, you should see a warning from direnv:
+
+```text
+direnv: WARNING: nix packages out of date. Run nix/update.sh
```
Comment (gidjin) [inline README.md:364]: This is exciting

File: README.md [hunk 2]
```diff
@@ -341,6 +345,25 @@ If you want to develop against AWS services you will need an AWS user account wi
 
 AWS credentials are managed via `aws-vault`. Once you have received AWS credentials (which are provided by the infrastructure team), you can follow these instructions to [finish setting up AWS](https://github.com/transcom/transcom-infrasec-gov/blob/master/docs/runbook/0001-aws-organization-authentication.md).
 
+### Setup: Nix
+
+NOTE: Nix is an experiment. If you are setting things up with Nix you
```
Comment (gidjin) [inline README.md:350]: I wonder if the note should also say "you may need to do a bit more digging to support nix than other systems See the #code-nix channel for help"?

**Review by gidjin — APPROVED** [review]
Changes look good. This looks fairly straight forward to setup! I left a comment to see if we should clarify the NOTE about it being an experiment. Other than that looks good to me. 

I was able to run the update, server tests, client tests, and start the servers. Might be good to have someone who doesn't have nix also try it out, just in case. But I don't see how it would be a problem.

File: README.md (hunk 2 above)
Comment (ahobson) [reply README.md:350]: Yeah, great point! I added some text about that.

**Pull request merged**